model/llm/prompt/parse_test.go

```diff
@@ -107,16 +107,29 @@ func TestParseResponse(t *testing.T) {
 			ExpectedCode: code,
 		})
 
-		validate(t, &testCase{
-			Name: "With Prose",
-
-			Response: "Some text...\n\n```\n" + code + "\n```\n\nSome more text...",
-
-			ExpectedAssessment: metrics.Assessments{
-				metrics.AssessmentKeyResponseNoExcess: 0,
-				metrics.AssessmentKeyResponseWithCode: 1,
-			},
-			ExpectedCode: code,
+		t.Run("Prose", func(t *testing.T) {
+			validate(t, &testCase{
+				Name: "After Newline",
+
+				Response: "Some text...\n\n```\n" + code + "\n```\n\nSome more text...",
+
+				ExpectedAssessment: metrics.Assessments{
+					metrics.AssessmentKeyResponseNoExcess: 0,
+					metrics.AssessmentKeyResponseWithCode: 1,
+				},
+				ExpectedCode: code,
+			})
+			validate(t, &testCase{
+				Name: "No Newline",
```

Comment:
The examples of the issue have `.` right before the code fence. Is this a typo, or actually the response we get? If so, there is no test case for that, right? Like no new line before the code fence.

Reply:
I used that to mark the space character because it is very easy to miss. It should say so in the issue. The actual example had a space before the first fence, and we already have that example.